GeneralsMD/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -97,6 +97,40 @@
 // ------------------------------------------------------------------------------------------------
 static const RGBColor IllegalBuildColor = { 1.0, 0.0, 0.0 };
 
+// ------------------------------------------------------------------------------------------------
+static UnicodeString formatMoneyValue(UnsignedInt amount)
+{
+	UnicodeString result;
+	if (amount >= 100000)
+	{
+		result.format(L"%uk", amount / 1000);
+	}
+	else
+	{
+		result.format(L"%u", amount);
+	}
+	return result;
+}
+
+static UnicodeString formatIncomeValue(UnsignedInt cashPerMin)
+{
+	UnicodeString result;
+	if (cashPerMin >= 10000)
+	{
+		result.format(L"%uk", cashPerMin / 1000);
+	}
+	else if (cashPerMin >= 1000)
+	{
+		UnsignedInt k = cashPerMin / 100;
+		result.format(L"%u.%uk", k / 10, k % 10);
```

Comment:
How about we scrap this format and just have "10k" for >= 10000 and "1000" for <10000. Because "1.2k" has 4 characters just like "1000" does, but is more complicated to look at.

I tested the following format and that looked good and simple:

```c++
static UnicodeString formatIncomeValue(UnsignedInt cashPerMin)
{
	UnicodeString result;
	if (cashPerMin >= 10000)
	{
		result.format(L"%uk", cashPerMin / 1000);
	}
	else if (cashPerMin >= 1000)
	{
		result.format(L"%u", (cashPerMin / 100) * 100);
	}
	else
	{
		result.format(L"%u", (cashPerMin / 10) * 10);
	}
	return result;
}
```